### PR TITLE
fix #1312: stem extension calculation fixed

### LINKED ISSUE #1312: StaveNote StaveNote_BoundingBoxes___Treble Test - Visual Error
Wrong bounding box with Bravura and Petaluma
StaveNote StaveNote_BoundingBoxes___Treble Bravura
StaveNote StaveNote_BoundingBoxes___Treble Petaluma

### PR DESCRIPTION
**Pull request by rvilarl**
fixes #1312

### CODE REVIEW

**Comment by rvilarl** [comment]
Visual differences to master:

Boundig box fixed (not picked by the visual difference script):
**Current**
pptr-StaveNote StaveNote_BoundingBoxes___Treble Bravura
**Reference**
pptr-StaveNote StaveNote_BoundingBoxes___Treble Bravura

Some minor differences due to the length of stem: 
**Current**
pptr-canvas_Grace_Notes Grace_Note_Stem Bravura_current
**Reference**
pptr-canvas_Grace_Notes Grace_Note_Stem Bravura_reference
**Current**
pptr-canvas_StaveNote Flag_and_Dot_Placement___Stem_Up Bravura_current
**Reference**
pptr-canvas_StaveNote Flag_and_Dot_Placement___Stem_Up Bravura_reference
Brackets now not overlapping as a result of the right calculation of the bounding box:
**Current**
pptr-Formatter Proportional_Formatting___No_Justification Petaluma svg_current
**Reference**
pptr-Formatter Proportional_Formatting___No_Justification Petaluma svg_reference
**Current**
pptr-Formatter Proportional_Formatting___No_Tuning Bravura svg_current
**Reference**
pptr-Formatter Proportional_Formatting___No_Tuning Bravura svg_reference

File: src/stavenote.ts [hunk 1]
```diff
@@ -1089,6 +1096,10 @@ export class StaveNote extends StemmableNote {
     ctx.closeGroup();
   }
 
+  getStaveNoteScale(): number {
```
Comment (0xfe) [inline src/stavenote.ts:1099]: Can you add a comment here that this is primarily used as the scaling factor for grace notes?

**Review by 0xfe — COMMENTED** [review]
Overall, looks excellent. One minor comment.

File: src/stavenote.ts (hunk 1 above)
Comment (rvilarl) [reply src/stavenote.ts:1100]: Done

**Comment by rvilarl** [comment]
Ready to merge

**Comment by 0xfe** [comment]
Looks good. Merging.

**Pull request merged**